pkg/controller/perconaservermongodbbackup/perconaservermongodbbackup_controller.go

```diff
@@ -341,16 +341,39 @@ func (r *ReconcilePerconaServerMongoDBBackup) checkFinalizers(ctx context.Contex
 						return errors.Wrap(err, "get storage")
 					}
 					if err := dummyPBM.DeleteBackupFiles(getPBMBackupMeta(cr), stg); err != nil {
-						log.Error(err, "failed to run finalizer", "finalizer", f)
-						finalizers = append(finalizers, f)
-					}
-				} else {
-					e := b.pbm.C.Logger().NewEvent(string(pbm.CmdDeleteBackup), "", "", primitive.Timestamp{})
-					err = b.pbm.C.DeleteBackup(cr.Status.PBMname, e)
-					if err != nil {
-						log.Error(err, "failed to run finalizer", "finalizer", f)
+						log.Error(err, "failed to run finalizer with dummy pbm", "finalizer", f)
 						finalizers = append(finalizers, f)
 					}
+					continue
+				}
+
+				if cluster == nil {
+					return errors.Errorf("PerconaServerMongoDB %s is not found", cr.Spec.GetClusterName())
+				}
+
+				var storage psmdbv1.BackupStorageSpec
+				switch {
+				case cr.Status.S3 != nil:
+					storage.Type = psmdbv1.BackupStorageS3
+					storage.S3 = *cr.Status.S3
+				case cr.Status.Azure != nil:
+					storage.Type = psmdbv1.BackupStorageAzure
+					storage.Azure = *cr.Status.Azure
+				}
+
+				priorities, err := b.pbm.GetPriorities(ctx, r.client, cluster)
+				if err != nil {
+					return errors.Wrap(err, "get priorities")
+				}
+				err = b.pbm.SetConfig(ctx, storage, b.spec.PITR, priorities)
```

Comment:
So if I understand correctly we set storage to the one which we need to delete, do we set it back to the one which was before?

Reply:
I don't think that we should set it to the previous one. When we do a restore, we set a proper config here: https://github.com/percona/percona-server-mongodb-operator/blob/1235fb3b92af16e9fc2691e5a2a5a2ceb882e228/pkg/controller/perconaservermongodbrestore/perconaservermongodbrestore_controller.go#L228

Reply:
> I don't think that we should set it to the previous one. When we do a restore, we set a proper config here:
> 
> https://github.com/percona/percona-server-mongodb-operator/blob/1235fb3b92af16e9fc2691e5a2a5a2ceb882e228/pkg/controller/perconaservermongodbrestore/perconaservermongodbrestore_controller.go#L228

But what about the backup?

Reply:
we set config before backup too: https://github.com/percona/percona-server-mongodb-operator/blob/main/pkg/controller/perconaservermongodbbackup/backup.go#L48